src/SwiftUI/SwiftUI/Views/ViewModifier.cs

```diff
@@ -0,0 +1,11 @@
+using System;
+using Swift.Interop;
+
+namespace SwiftUI
+{
+	[SwiftImport (SwiftUILib.Path)]
+	public abstract unsafe record ViewModifier<T> : View
```

Comment:
I know this is still WIP, but `ViewModifier` should not inherit from `View` and it shouldn't be generic. It's so similar to `View` that we can probably copy most of the implementation from there, and also create a `CustomViewModifierType` that's mostly a copy of `CustomViewType`.

Reply:
@chkn Just taking a look at your suggestion. Shouldn't we maybe have a `BaseView` which both `ViewModifier` inherit from so that we aren't duplicating the `AddRef()`, `UnRef ()` etc? If the memory allocation etc is similar between `CustomViewType` and `CustomViewModifierType`, maybe something similar there too. Thoughts?

Reply:
Yeah absolutely!

For `CustomViewType`, we already have the base class, `ManagedSwiftType`. I think everything that's left in `CustomViewType` is pretty specific to `View`. `ViewModifier` has a very similar shape to `View`, but you would still need to copy and change some things, like `ViewProtocolConformanceDescriptor` -> `ViewModifierProtocolConformanceDescriptor` and changing the reqs to the appropriate ones from the `ViewModifier` protocol (https://github.com/chkn/Xamarin.SwiftUI/blob/main/src/SwiftUI/SwiftUI/Interop/ViewProtocolConformanceDescriptor.cs#L22-L31). It would be amazing to figure out how to abstract some of that, especially since the goal is to auto-generate as much of this code as possible in the future. I'm fine with basically duplicating it for now, but if you have some inspiration, feel free to experiment.

The `View` side seems like lower hanging fruit--it definitely seems like we could have something like `ManagedSwiftStruct` that derives from `SwiftStruct` and holds all the `AddRef`, `UnRef`, etc.

Reply:
`ManagedSwiftStruct` is a much nicer base name 🙂